napari/components/_tests/test_layer_slicer.py

```diff
@@ -58,16 +58,19 @@ def __call__(self) -> FakeSliceResponse:
 
 class FakeAsyncLayer:
     def __init__(self):
-        self.slice_count = 0
-        self.lock = RLock()
+        self.async_slice_count: int = 0
+        self.sync_slice_count: int = 0
+        self.lock: RLock = RLock()
 
-    def _make_slice_request(self, dims) -> FakeSliceRequest:
+    def _make_slice_request(self, dims: Dims) -> FakeSliceRequest:
         assert current_thread() == main_thread()
-        self.slice_count += 1
-        return FakeSliceRequest(id=self.slice_count, lock=self.lock)
+        return FakeSliceRequest(id=self.async_slice_count + 1, lock=self.lock)
```

Comment:
I made this change, but it's not quite right since it doesn't deal with multiple async slice requests properly.